prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5322,31 +5322,75 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
+            mir::NullOp::Box => self.encode_assign_box(
+                op_ty,
+                encoded_lhs,
+                ty,
+                location,
+            ),
+            mir::NullOp::SizeOf => {
+                // TODO: ParamEnv::empty() should probably be tyctxt.param_env(def_id_of_method)
+                let bytes = self.encoder.env().tcx()
+                    .layout_of(ParamEnv::empty().and(op_ty))
+                    .unwrap().layout
+                    .size
+                    .bytes();
+                let bytes_vir = vir::Expr::from(bytes);
+                self.encode_copy_value_assign(
                     encoded_lhs,
-                    ref_field,
+                    bytes_vir,
+                    ty,
                     location,
-                    vir::AssignKind::Move,
-                    false
-                )?;
+                )
+            },
+            mir::NullOp::AlignOf => {
+                // TODO: ParamEnv::empty() should probably be tyctxt.param_env(def_id_of_method)
+                let bytes = self.encoder.env().tcx()
+                    .layout_of(ParamEnv::empty().and(op_ty))
+                    .unwrap().layout
+                    .align.abi // FIXME: abi or pref?
+                    .bytes();
+                let bytes_vir = vir::Expr::from(bytes);
+                self.encode_copy_value_assign(
+                    encoded_lhs,
+                    bytes_vir,
+                    ty,
+                    location,
+                )
+            },
+        }
+    }
 
-                // Allocate `box_content`
-                stmts.extend(self.encode_havoc_and_allocation(&box_content));
+    fn encode_assign_box(
+        &mut self,
+        op_ty: ty::Ty<'tcx>,
+        encoded_lhs: vir::Expr,
+        ty: ty::Ty<'tcx>,
+        location: mir::Location,
+    ) -> SpannedEncodingResult<Vec<vir::Stmt>> {
+        assert_eq!(op_ty, ty.boxed_ty());
+        let ref_field = self.encoder.encode_dereference_field(op_ty)
+            .with_span(
+                self.mir_encoder.get_span_of_location(location)
+            )?;
+        let box_content = encoded_lhs.clone().field(ref_field.clone());
 
-                // Leave `box_content` uninitialized
-                Ok(stmts)
-            }
-            mir::NullOp::SizeOf => unimplemented!(),
-            mir::NullOp::AlignOf => unimplemented!(),
-        }
+        let mut stmts = self.prepare_assign_target(
+            encoded_lhs,
+            ref_field,
+            location,
+            vir::AssignKind::Move,
+            false
+        )?;
+
+        // Allocate `box_content`
+        // TODO: This is also encoding initialization, which should be avoided because both
+        //   `Rvalue::NullaryOp(NullOp::Box)` and `Rvalue::ShallowInitBox` don't initialize the
+        //   content of the box.
+        stmts.extend(self.encode_havoc_and_initialization(&box_content));
```

Comment:
We have to plan how to solve this. If we don't inhale the permissions here, then every assignment to a box should assume that the box is potentially uninitialized (or check if it has already been initialized), which would be less efficient for the verifier.